cfme/utils/appliance/__init__.py

```diff
@@ -102,7 +102,43 @@ class ApplianceException(Exception):
     pass
 
 
-class IPAppliance(object):
+class BaseAppliance(object):
+    """ Represents base appliance class intended to hold base properties existing in every appliance
+    """
+    type = None
+    CONFIG_MAPPING = {
+        'hostname': 'hostname',
+        'ui_protocol': 'ui_protocol',
+        'ui_port': 'ui_port',
+        'browser_steal': 'browser_steal',
+        'container': 'container',
+        'pod': 'container',
+        'openshift_creds': 'openshift_creds',
+        'is_dev': 'is_dev',
+        'db_host': 'db_host',
+        'db_port': 'db_port',
+        'ssh_port': 'ssh_port',
+        'project': 'project',
+        'version': 'version',
+        'type': 'type'
+    }
+    CONFIG_NONGLOBAL = {'hostname'}
+
+    @property
+    def as_json(self):
+        """Dumps the arguments that can create this appliance as a JSON. None values are ignored."""
+        def _version_tostr(x):
+            if isinstance(x, Version):
+                return str(x)
+            else:
+                return x
+        return json.dumps({
+            k: _version_tostr(getattr(self, k))
+            for k in set(self.CONFIG_MAPPING.values())
+            if hasattr(self, k) and k != "collections"})
```

Comment:
Q: I'm confused, why we are caring about `collections` here? Mapping not include collection attr right?

Reply:
yes, serialized appliance shouldn't include collections